model-optimizer/extensions/ops/range.py

```diff
@@ -70,6 +70,9 @@ def infer(node: Node):
         start = node.in_port(0).data.get_value()
         limit = node.in_port(1).data.get_value()
         delta = node.in_port(2).data.get_value()
+        if any(np.isinf((start, limit, delta))):
+            raise Error("Range node's '{}' input values must be finite, but instead "
+                        "it contain infinities: start={}, stop={}, step={}".format(name, start, limit, delta))
```

Comment:
Yes this can happen, if e.g. stop=np.array(100000, dtype=np.float32) after casting to fp16 `100000` -> `+inf`. If we pass this value to numpy it will fail with unexpected error and user will not have any info at which node error happend.

Therefore should raise an user friendly Error for Range in such cases as soon as we faced infinities before numpy